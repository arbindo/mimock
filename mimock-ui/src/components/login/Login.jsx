import React from 'react';
import { LoginWrapper, InnerFlex, FormContainer } from './Login.style';
import SocialWrapper from './SocialWrapper';
import LoginForm from './LoginForm';

export default function Login() {
	return (
		<LoginWrapper>
			<InnerFlex>
				<SocialWrapper />
				<FormContainer>
					<LoginForm />
				</FormContainer>
			</InnerFlex>
		</LoginWrapper>
	);
}
