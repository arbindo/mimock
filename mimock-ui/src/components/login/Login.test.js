import React from 'react';
import { render } from '@testing-library/react';
import Login from './Login';

const mockedNavigation = jest.fn(() => {});
jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockedNavigation,
}));

const mockedCookieSet = jest.fn();
jest.mock('react-cookie', () => ({
	Cookies: jest.fn().mockImplementation(() => ({
		set: mockedCookieSet,
		get: () => jest.fn(),
	})),
}));

describe('Login', () => {
	it('should enter login component', async () => {
		const tree = await render(<Login />);

		const { container, getByTestId, queryByTestId } = tree;

		expect(getByTestId('login-social-wrapper')).toBeInTheDocument();
		expect(getByTestId('login-form')).toBeInTheDocument();

		expect(queryByTestId('login-error-label')).not.toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
