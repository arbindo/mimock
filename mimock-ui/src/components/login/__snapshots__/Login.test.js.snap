// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should enter login component 1`] = `
<div>
  <div
    class="flex w-full h-full mx-auto my-auto items-center align-middle sc-bdvvtL EBYGm"
  >
    <div
      class="flex flex-row w-2/4 xl:w-2/4 lg:w-2/3 md:w-11/12 sm:w-11/12 rounded-md shadow-md border-2 border-gray-50 bg-white mx-auto my-auto"
    >
      <div
        class="block w-1/2 h-full justify-center mx-auto rounded-tl-md rounded-bl-md border-r-2 bg-white dark:bg-slate-600"
        data-testid="login-social-wrapper"
      >
        <div
          class="block w-full h-11/12"
        >
          <div
            class="flex items-center justify-center mx-auto my-10 bg-no-repeat bg-center bg-contain sc-gsDKAQ jepqxS"
            data-testid="login-social-logo"
          />
          <h1
            class="mt-4 mx-4 text-left font-bold text-2xl font-sans text-gray-700 dark:text-gray-300"
            data-testid="login-social-title"
          >
            MIMOCK
          </h1>
          <h1
            class="mx-4 text-left font-light font-sans text-gray-700 dark:text-gray-300"
            data-testid="login-social-description"
          >
            PLATFORM TO MOCK REST ENDPOINTS
          </h1>
          <div
            class="flex justify-around items-center align-middle w-1/2 mx-auto my-14 text-4xl text-slate-800 dark:text-gray-300 hover:cursor-pointer"
            data-testid="login-social-icons"
          >
            <a
              class=""
              data-testid="slack-logo"
              href="https://mimock.slack.com"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M94.12 315.1c0 25.9-21.16 47.06-47.06 47.06S0 341 0 315.1c0-25.9 21.16-47.06 47.06-47.06h47.06v47.06zm23.72 0c0-25.9 21.16-47.06 47.06-47.06s47.06 21.16 47.06 47.06v117.84c0 25.9-21.16 47.06-47.06 47.06s-47.06-21.16-47.06-47.06V315.1zm47.06-188.98c-25.9 0-47.06-21.16-47.06-47.06S139 32 164.9 32s47.06 21.16 47.06 47.06v47.06H164.9zm0 23.72c25.9 0 47.06 21.16 47.06 47.06s-21.16 47.06-47.06 47.06H47.06C21.16 243.96 0 222.8 0 196.9s21.16-47.06 47.06-47.06H164.9zm188.98 47.06c0-25.9 21.16-47.06 47.06-47.06 25.9 0 47.06 21.16 47.06 47.06s-21.16 47.06-47.06 47.06h-47.06V196.9zm-23.72 0c0 25.9-21.16 47.06-47.06 47.06-25.9 0-47.06-21.16-47.06-47.06V79.06c0-25.9 21.16-47.06 47.06-47.06 25.9 0 47.06 21.16 47.06 47.06V196.9zM283.1 385.88c25.9 0 47.06 21.16 47.06 47.06 0 25.9-21.16 47.06-47.06 47.06-25.9 0-47.06-21.16-47.06-47.06v-47.06h47.06zm0-23.72c-25.9 0-47.06-21.16-47.06-47.06 0-25.9 21.16-47.06 47.06-47.06h117.84c25.9 0 47.06 21.16 47.06 47.06 0 25.9-21.16 47.06-47.06 47.06H283.1z"
                />
              </svg>
            </a>
            <a
              class=""
              data-testid="gh-logo"
              href="https://github.com/arbindo/mimock"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 496 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                />
              </svg>
            </a>
          </div>
        </div>
        <h1
          class="text-right font-sans text-xs text-gray-400 mr-1 my-2 h-1/12"
          data-testid="login-social-license-label"
        >
          Apache-2.0 Licensed
        </h1>
      </div>
      <div
        class="w-1/2 bg-teal-50 p-6"
      >
        <div
          class="w-11/12 h-full flex flex-col mx-auto my-auto justify-center align-middle"
          data-testid="login-form"
        >
          <div
            class="font-sans text-xl font-bold text-gray-700"
          >
            Login to mimock
          </div>
          <div
            class="bg-teal-500 w-36 h-1.5"
          />
          <form
            class="w-11/12 my-5 mx-auto flex flex-col justify-center"
          >
            <div
              class="font-sans text-gray-500 mt-3"
              data-testid="login-username-label"
            >
              USER NAME
            </div>
            <input
              autocomplete="off"
              class="w-full my-2 p-2 outline-none rounded-md border-2 border-gray-300"
              data-testid="login-username-input"
              placeholder="Enter username"
              required=""
              value=""
            />
            <div
              class="font-sans text-gray-500 mt-3"
              data-testid="login-password-label"
            >
              PASSWORD
            </div>
            <input
              autocomplete="off"
              class="w-full my-2 p-2 outline-none rounded-md border-2 border-gray-300 sc-dkPtRN"
              data-testid="login-password-input"
              placeholder="Enter password"
              required=""
              type="password"
              value=""
            />
            <button
              class="px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 mt-5 text-white w-full bg-teal-500"
              data-testid="login-submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
